tests/test_compress.py

```diff
@@ -504,3 +504,86 @@ def test_compress_numpy_data(tmp_path):
 
     compress_numpy_and_compare_data(17, 0, 65, 8, 65, tmp_path, 8, 1e-2, blockshape=(32, 32, 4))
     compress_numpy_and_compare_data(801, 0, 9, 8, 13, tmp_path, 8, 1e-2, blockshape=(16, 16, 16))
+
+
+def compress_stream_and_compare_data(
+    n_samples,
+    min_iline,
+    n_ilines,
+    min_xline,
+    n_xlines,
+    tmp_path,
+    bits_per_voxel,
+    rtol,
+    blockshape=(4, 4, -1),
+):
+
+    out_sgz = os.path.join(str(tmp_path), "from-stream.sgz")
+    out_sgz_no_headers = os.path.join(str(tmp_path), "from-stream_no_headers.sgz")
+
+    array, ilines, xlines, samples = generate_fake_seismic(
+        n_ilines, n_xlines, n_samples, min_iline=min_iline, min_xline=min_xline
+    )
+
+    trace_headers = {
+        segyio.tracefield.TraceField.INLINE_3D: np.broadcast_to(
+            np.expand_dims(ilines, axis=1), (n_ilines, n_xlines)
+        ),
+        segyio.tracefield.TraceField.CROSSLINE_3D: np.broadcast_to(
+            xlines, (n_ilines, n_xlines)
+        ),
+    }
+
+    with StreamConverter(
+        out_sgz,
+        ilines=ilines,
+        xlines=xlines,
+        samples=samples,
+        bits_per_voxel=bits_per_voxel,
+        blockshape=blockshape,
+        trace_headers=trace_headers,
+    ) as converter:
+        for i in range(0, array.shape[0], blockshape[0]):
+            end = min(i + blockshape[0], array.shape[0])
+            chunk = array[i:end, :, :]
+            converter.write(chunk)
+
+    with SgzReader(out_sgz) as reader:
+        assert np.allclose(reader.ilines, ilines, rtol=rtol)
+        assert np.allclose(reader.xlines, xlines, rtol=rtol)
+        assert np.allclose(reader.read_volume(), array, rtol=rtol)
+        assert 20 == reader.get_file_source_code()
```

Comment:
Maybe you want to define this 20 somewhere?

Reply:
actually I don't know what it is. since StreamConverter does the 'same' thing as NumpyConverter I just borrowed the checks from there